clang/lib/Driver/Driver.cpp

```diff
@@ -3853,6 +3864,53 @@ class OffloadingActionBuilder final {
       SYCLDeviceActions.clear();
     }
 
+    void addSYCLDeviceLibs(const ToolChain *TC, ActionList &DeviceLinkObjects,
+                           bool isSpirvAOT, bool isMSVCEnv) {
+      enum SYCLDeviceLibType {
+        sycl_devicelib_wrapper,
+        sycl_devicelib_fallback
+      };
+      StringRef LibLoc, LibSysUtils;
+      if (isMSVCEnv) {
+        LibLoc = Args.MakeArgString(TC->getDriver().Dir + "/../bin");
+        LibSysUtils = "libsycl-msvc";
+      } else {
+        LibLoc = Args.MakeArgString(TC->getDriver().Dir + "/../lib");
+        LibSysUtils = "libsycl-glibc";
+      }
+      SmallVector<StringRef, 4> sycl_device_wrapper_libs = {
+          LibSysUtils, "libsycl-complex", "libsycl-complex-fp64",
+          "libsycl-cmath", "libsycl-cmath-fp64"};
+      // For AOT compilation, we need to link sycl_device_fallback_libs as
+      // default too.
+      SmallVector<StringRef, 4> sycl_device_fallback_libs = {
+          "libsycl-fallback-cassert", "libsycl-fallback-complex",
+          "libsycl-fallback-complex-fp64", "libsycl-fallback-cmath",
+          "libsycl-fallback-cmath-fp64"};
+      auto addInputs = [&](SYCLDeviceLibType t) {
+        auto sycl_libs = (t == sycl_devicelib_wrapper)
+                             ? sycl_device_wrapper_libs
+                             : sycl_device_fallback_libs;
+        for (const StringRef &Lib : sycl_libs) {
+          SmallString<128> LibName(LibLoc);
+          llvm::sys::path::append(LibName, Lib);
+          llvm::sys::path::replace_extension(LibName, ".o");
```

Comment:
`.obj` for Windows?

Reply:
Hi, @mdtoguchi 
Once the https://github.com/intel/llvm/pull/2405 is approved and merged, I will rebase the code and align with the new library suffix and libsycl-crt.o
Thank you very much.